File: Api/Models/ProviderOrder.cs
namespace Api.Models;

public class ProviderOrder : Auditable
{
    public int Id { get; set; }
    
    public DateTime Date { get; set; }
        
    public int ProviderId { get; set; }
    public Provider? Provider { get; set; }
    
    public int StatusId { get; set; }
    public Status? Status { get; set; }
        
    public virtual List<ArticleOrder>? ArticleOrders { get; set; }
}

File: WebApp/src/app/models/providerOrder.ts
import {Provider} from "./provider";
import {Status} from "./status";
import {ArticleOrder} from "./article";
import {Auditable} from "./auditable";

export interface ProviderOrder extends Auditable{
  id?: number;
  date?: Date;
  providerId?: number;
  provider?: Provider;
  statusId?: number;
  status?: Status;
  articleOrders?: Array<ArticleOrder>;
}
